OpenRA.Mods.Common/Effects/RallyPointIndicator.cs

```diff
@@ -53,15 +75,21 @@ public void Tick(World world)
 
 		public IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
-			if (building.Owner != building.World.LocalPlayer)
-				return SpriteRenderable.None;
+			if (!building.IsInWorld || !building.Owner.IsAlliedWith(building.World.LocalPlayer))
+				yield break;
 
-			if (!building.IsInWorld || !building.World.Selection.Actors.Contains(building))
-				return SpriteRenderable.None;
+			if (!building.World.Selection.Actors.Contains(building))
+				yield break;
```

Comment:
Please continue to use `return SpriteRenderable.None` here - it's more efficient. The actual rendering code can continue to use `yield` if you create another method for it.